festim/boundary_conditions/flux_bc.py

```diff
@@ -0,0 +1,149 @@
+import festim as F
+import ufl
+from dolfinx import fem
+import numpy as np
+
+
+class FluxBC:
+    """
+    Flux boundary condition class
+
+    Ensuring the gradient of the solution:
+    -D * grad(c) * n = f
+
+    Args:
+        subdomain (festim.SurfaceSubdomain): the surface subdomain where the boundary
+            condition is applied
+        value (float or fem.Constant): the value of the boundary condition
+        species (festim.Species): the species to which the condition is applied
+
+    Attributes:
+        subdomain (festim.SurfaceSubdomain): the surface subdomain where the boundary
+            condition is applied
+        value (float or fem.Constant): the value of the boundary condition
+        species (festim.Species): the species to which the condition is applied
+        value_fenics (fem.Function or fem.Constant): the value of the boundary condition in
+            fenics format
+        bc_expr (fem.Expression): the expression of the boundary condition that is used to
+            update the value_fenics
+
+
+    Usage:
+        >>> from festim import FluxBC
+        >>> FluxBC(subdomain=my_subdomain, value=1, species="H")
+        >>> FluxBC(subdomain=my_subdomain, value=lambda x: 1 + x[0], species="H")
+        >>> FluxBC(subdomain=my_subdomain, value=lambda t: 1 + t, species="H")
+        >>> FluxBC(subdomain=my_subdomain, value=lambda T: 1 + T, species="H")
+        >>> FluxBC(subdomain=my_subdomain, value=lambda x, t: 1 + x[0] + t, species="H")
+    """
+
+    def __init__(self, subdomain, value, species) -> None:
+        self.subdomain = subdomain
+        self.value = value
+        self.species = species
+
+        self.value_fenics = None
+        self.bc_expr = None
+
+    @property
+    def value_fenics(self):
+        return self._value_fenics
+
+    @value_fenics.setter
+    def value_fenics(self, value):
+        if value is None:
+            self._value_fenics = value
+            return
+        if not isinstance(value, (fem.Function, fem.Constant, np.ndarray)):
+            raise TypeError(
+                f"Value must be a dolfinx.fem.Function, dolfinx.fem.Constant, or a np.ndarray not {type(value)}"
+            )
+        self._value_fenics = value
+
+    @property
+    def time_dependent(self):
+        if self.value is None:
+            return False
```

Comment:
Isn't this a bit dangerous? Say I check if it's time-dependent before assigning it a value by mistake, I may think it's not time-dpeendent but then it becomes time dependent if the value is say a function of time.

Should we raise an error instead?

Reply:
Yeah this is valid, but would be worth changing the other cases of this property in other classes too

Reply:
Opened an issue to track progress on this #744